### PR TITLE
feat: setup bundlesize

### LINKED ISSUE #829: Keeping track of bundle size
Size snapshot is only ran when you run

`yarn build`

This can lead to the size-snapshot.json getting out of sync.

Let's take an example.

* Person A adds new component to UI-Kit
* Person A works with storybook, and never runs `yarn build`
* Person A makes pull request
* Person A's new component is merged to master
* Person B makes a different contribute, but said person runs `yarn build`
* Now size-snapshot is ran and the json is updated, but the "real" changes come from both Person A's new component and Person B's work.

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Adds `bundlesize` to track our bundlesize in PRs

Closes #829

### CODE REVIEW

**Comment by tdeekens** [comment]
Huuzaaah.

**Comment by montezume** [comment]
I added the token to our CI variables. I'm not sure if anything is missing 🤔

File: bundlesize.config.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "files": [
+    {
+      "path": "./dist/ui-kit.esm.js",
+      "maxSize": "175 kB"
```
Comment (emmenko) [inline bundlesize.config.json]: Hmm are you sure this is correct?

https://bundlephobia.com/result?p=@commercetools-frontend/ui-kit@9.9.0

File: package.json [hunk 2]
```diff
@@ -122,6 +122,7 @@
     "babel-plugin-transform-rename-import": "2.3.0",
     "babel-preset-jest": "24.6.0",
     "browserslist": "4.6.3",
+    "bundlesize": "0.18.0",
```
Comment (emmenko) [inline package.json]: We also need to trigger the bundlesize on CI...

File: bundlesize.config.json (hunk 1 above)
Comment (montezume) [reply bundlesize.config.json]: Can you give more information? What is correct? The max size?

File: package.json (hunk 2 above)
Comment (montezume) [reply package.json]: how do we do this? I can't find instructions

Comment (montezume) [reply package.json]: I added the token to our env vars

Comment (emmenko) [reply package.json]: Just add a step on CI to execute `bundlesize`. That's it.

After this: https://github.com/commercetools/ui-kit/blob/20cd7a4f5a2b2b75c6e9052712631253a98a5c6b/.circleci/config.yml#L36

Just add 

```
- run: yarn run bundlesize
```

File: bundlesize.config.json (hunk 1 above)
Comment (emmenko) [reply bundlesize.config.json]: Yes the max size...currently it's like >700KB

File: .circleci/config.yml [hunk 3]
```diff
@@ -131,6 +135,12 @@ jobs:
       - install_dependencies
       - restore_workspace
       - bundle_test
+  bundle_size_test:
```
Comment (emmenko) [inline .circleci/config.yml]: No new job please, just keep it simple: https://github.com/commercetools/ui-kit/pull/914#discussion_r300316270

File: bundlesize.config.json (hunk 1 above)
Comment (montezume) [reply bundlesize.config.json]: No, it's correct. When I run bundlesize it gives ~165.

Comment (montezume) [reply bundlesize.config.json]: Screen Shot 2019-07-04 at 11 40 34 AM
which is very close to your links result...

File: package.json (hunk 2 above)
Comment (montezume) [reply package.json]: done

File: bundlesize.config.json (hunk 1 above)
Comment (emmenko) [reply bundlesize.config.json]: Ah ok it gzippes it

image

**Comment by montezume** [comment]
Should I add this to app kit as well? 🤔

File: bundlesize.config.json (hunk 1 above)
Comment (montezume) [reply bundlesize.config.json]: In fact I think we should lower this. It should be maybe 2-5%?

**Pull request merged**